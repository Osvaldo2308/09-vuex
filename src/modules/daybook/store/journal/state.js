export default ()=>({
    isLoading: true,
    entries: [
        /* {
            id: '1',
            date: new Date().toDateString(),
            text: 'Enim deserunt sit adipisicing magna non consequat ullamco. Aliquip ex aliquip laboris eu adipisicing incididunt consequat esse minim deserunt eiusmod est dolore ex. Eu magna et anim dolor. Est fugiat laborum quis cillum. Adipisicing sunt dolore nostrud ipsum culpa eiusmod velit. Magna qui ipsum in sint occaecat magna in magna. Amet aliquip do tempor cupidatat dolore incididunt Lorem sint id.',
            picture:null,
        },
        {
            id: '2',
            date: new Date().toDateString(),
            text: 'Anim fugiat ullamco proident dolor pariatur. Amet ea non consequat non magna mollit sunt in velit. Do ipsum elit mollit est in mollit fugiat quis ullamco cupidatat minim exercitation laboris.',
            picture:null,
        },
        {
            id: '3',
            date: new Date().toDateString(),
            text: 'Nisi sit ut consectetur adipisicing occaecat non aliqua. Tempor pariatur cupidatat dolor sit nulla laborum. Aliquip ipsum nulla pariatur proident incididunt consequat elit anim. Nisi commodo non proident duis do eu mollit labore esse elit.',
            picture:null,
        }, */
    ]
})