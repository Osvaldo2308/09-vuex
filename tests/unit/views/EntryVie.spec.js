import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'


import journal from '@/modules/daybook/store/journal'
import {journalState} from '../../../../07-journal-main/tests/unit/mock-data/test-journal-state'

import EntryView from "@/modules/daybook/views/EntryView.vue";

const createVuexStore = (initialState) =>
createStore ({
    modules:{
        journal:{
            ...journal,
            state:{ ...initialState}
        }
    }
})

describe('Pruebas en el Entry View', ()=>{

    const store = createVuexStore(journalState)
    const mockRouter = {
        push: jest.fn()
    }
    let wrapper

    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallowMount( EntryView,{
            props:{
                id: '-NNwAx5DxbG41eBMMnwO'
            },
            global:{
                mocks:{
                    $router: mockRouter
                },
            plugins: [ store ]
            }
        })
    })
    test('Debe de sacar al usuario por que el ID no existe', ()=>{
        const wrapper = shallowMount( EntryView,{
            props:{
                id: 'Este ID no existe en el STORE'
            },
            global:{
                mocks:{
                    $router: mockRouter
                },
            plugins: [ store ]
            }
        })

        expect (mockRouter.push) .toHaveBeenCalledWith({name: 'no-entry'})
    })

    test('Debe de mostrar la entrada correcta ', ()=>{
        expect(wrapper.html()).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalled()    
    })
})