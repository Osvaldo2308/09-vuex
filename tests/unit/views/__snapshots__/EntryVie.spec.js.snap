// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry View Debe de mostrar la entrada correcta  1`] = `
<div class="entry-title d-flex justify-content-between p-1">
  <div><span class="text-success fs-3 fw-bold">10</span><span class="mx-1 fs-3">Febrero</span><span class="mx-4 fs-4 fw-light">2023, Viernes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir Foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedio hoy?"></textarea></div>
<img src="https://res.cloudinary.com/dbpazexms/image/upload/v1676047011/sostymtaz3jbvfzpqrfi.png" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
